refarch-integrations/refarch-s3-integration/refarch-s3-integration-rest/refarch-s3-integration-rest-client-starter/src/main/java/de/muenchen/refarch/integration/s3/client/configuration/S3IntegrationClientAutoConfiguration.java

```diff
@@ -90,6 +88,7 @@ private WebClient authenticatedWebClient(
      */
     @Bean
     @ConditionalOnBean(SupportedFileExtensions.class)
+    @SuppressWarnings("PMD.LooseCoupling")
```

Comment:
General question. Can't we get rid of this error by changing to a more generous interface. This would be a way better solution instead of suppressing `PMD.LooseCoupling`. Also affects various parts of this PR.

Reply:
Not really know what you mean, but for beans switching to map is not possible in my opinion